pkg/systemd/hwinfo.jsx

```diff
@@ -65,20 +68,115 @@ const SystemInfo = ({ info }) => (
                 </React.Fragment>
             }
             { info.nproc !== undefined &&
-                <tr>
-                    <th>{ _("CPU") }</th>
-                    <td>{ (info.nproc > 1) ? `${info.nproc}x ${info.cpu_model}` : info.cpu_model }</td>
-                </tr>
+                <React.Fragment>
+                    <tr>
+                        <th>{ _("CPU") }</th>
+                        <td>{ (info.nproc > 1) ? `${info.nproc}x ${info.cpu_model}` : info.cpu_model }</td>
+                    </tr>
+                    <tr>
+                        <th>{ _("CPU Security") }</th>
+                        <td>{ onSecurityClick !== undefined && <a role="link" onClick={onSecurityClick}> Mitigations </a> }</td>
+                    </tr>
+                </React.Fragment>
             }
         </tbody>
     </table>
 );
 
+class CPUSecurityMitigationsDialog extends React.Component {
+    constructor(props) {
+        super(props);
+        this.close = this.close.bind(this);
+        this.saveAndReboot = this.saveAndReboot.bind(this);
+        this.state = {
+            nosmt: undefined,
+        };
+        this.nosmtRow();
+    }
+
+    close() {
+        if (this.props.onClose)
+            this.props.onClose();
+    }
+
+    nosmtRow() {
+        /* nosmt */
+        let promises = [cockpit.spawn(["lscpu", "--json"]), cockpit.spawn(["cat", "/proc/cmdline"])];
+        Promise.all(promises).then(values => {
+            let threads_per_core = Number(JSON.parse(values[0]).lscpu.filter(value => value.field.indexOf("Thread(s) per core") !== -1)[0].data);
+            let smt_disabled = values[1].indexOf("nosmt") !== -1;
+            if (values[1].indexOf("nosmt=") === -1 && (threads_per_core > 1 || (threads_per_core === 1 && smt_disabled)))
+                this.setState({ nosmt: smt_disabled });
+        });
+    }
+
+    saveAndReboot() {
+        this.setState(state => {
+            let options = [];
+            let promises = [];
+
+            if (state.nosmt)
+                options.push(['set', 'nosmt']);
+            else if (state.nosmt !== undefined)
+                options.push(['remove', 'nosmt']);
+
+            for (let option of options)
+                promises.push(cockpit.script(kernelopt_sh, option, { "superuser": true, "err": "message" }));
+
+            Promise.all(promises)
+                    .then(() => {
+                        cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: true, err: "message" });
+                    })
+                    .catch(() => {
+                        this.setState({ rebooting: false });
+                    });
+        });
+    }
+
+    render() {
+        let rows = [];
+
+        if (this.state.nosmt !== undefined)
+            rows.push((<ListingRow key="nosmt" columns={ [(<b>{ _("Disable symmetric multithreading") } (nosmt)</b>),
```

Comment:
This is entirely optional: In the long run we try to move away from our Cockpit listings, and use 
[PatternFly ListView](https://rawgit.com/patternfly/patternfly-react/gh-pages/patternfly-3/index.html?selectedKind=patternfly-react%2FContent+Views%2FList+View). These are really pleasant to use, I tried them in a first version of my update snapshots work. It had some bugs which ruled it out for that, but these wouldn't affect this list.

Reply:
This should also have a "Read more.." link to https://access.redhat.com/security/vulnerabilities/L1TF (for now), see the design. (There it is at the top, but it needs to be per-row).

Reply:
Is there a way to make this a `<th></th>` instead of a `<td><b></b></td>`?